### PR TITLE
Low audit issues

### PR DESCRIPTION
**Pull request by nuevoalex**
Includes:

#4957 
#4958 
#4967 
#5067 
#5097 
#5070 
#5075 
#5107

### CODE REVIEW

**Review by epheph — APPROVED** [review]
These changes look great. There are 3 more `msg.sender == address(this)` in Orders.sol (did i identify those correctly?) for #5107  .

**Comment by nuevoalex** [comment]
@epheph yea those are actually required for the TestOrders.sol contract to function which we use to do automated tests for the logic in storing orders

File: packages/augur-core/source/contracts/reporting/ReputationToken.sol [hunk 1]
```diff
@@ -35,6 +35,8 @@ contract ReputationToken is VariableSupplyToken, IV2ReputationToken {
         parentUniverse = _parentUniverse;
         warpSync = _augur.lookup("WarpSync");
         legacyRepToken = IERC20(augur.lookup("LegacyReputationToken"));
+        require(warpSync != address(0));
+        require(legacyRepToken != IERC20(0));
         erc1820Registry = IERC1820Registry(_erc1820RegistryAddress);
```
Comment (MicahZoltu) [inline packages/augur-core/source/contracts/reporting/ReputationToken.sol:40]: Need one for `erc1820Registry` as well, plus the same thing in several other contracts.

File: packages/augur-core/source/contracts/reporting/Universe.sol [hunk 2]
```diff
@@ -93,11 +93,24 @@ contract Universe is IUniverse {
...
+        require(marketFactory != IMarketFactory(0));
+        require(disputeWindowFactory != IDisputeWindowFactory(0));
+        require(shareToken != IShareToken(0));
+        require(repPriceOracle != IRepPriceOracle(0));
+        require(formulas != IFormulas(0));
+        require(cash != ICash(0));
+        require(daiVat != IDaiVat(0));
+        require(daiPot != IDaiPot(0));
+        require(daiJoin != IDaiJoin(0));
+    }
```
Comment (MicahZoltu) [inline packages/augur-core/source/contracts/reporting/Universe.sol:112]: Out of curiosity, why did you break this out into a separate function?  It feels like splitting it like this is going to increase the chances of missing/forgetting one vs something like:
```solidity
marketFactory = IMarketFactory(augur.lookup("MarketFactory"))
require(marketFactory != IMarketFactory(0);
```
Also, if you wanted to save a little gas by using a local variable:
```solidity
address _temp;
_temp = augur.lookup("MarketFactory");
require(_temp != address(0));
marketFactory = IMarketFactory(_temp);
_temp = augur.lookup("DisputeWindowFactory");
require(_temp != address(0));
disputeWindowFactory = IDisputeWindowFactory(_temp);
...
```

**Review by MicahZoltu — APPROVED** [review]

File: packages/augur-core/source/contracts/reporting/ReputationToken.sol (hunk 1 above)
Comment (epheph) [reply packages/augur-core/source/contracts/reporting/ReputationToken.sol:40]: It enters erc1820Registry on the next line, there is an implicit check (technically, on extcodesize != 0).

Comment (MicahZoltu) [reply packages/augur-core/source/contracts/reporting/ReputationToken.sol:40]: Good point @epheph.  However, there is still the same issue in multiple other contracts.  Not sure if @nuevoalex would prefer a bunch of small PRs, or a list in GitHub, or to just hunt them down yourself?

File: packages/augur-core/source/contracts/reporting/Universe.sol (hunk 2 above)
Comment (nuevoalex) [reply packages/augur-core/source/contracts/reporting/Universe.sol:112]: Thought this was more readable.

File: packages/augur-core/source/contracts/reporting/ReputationToken.sol (hunk 1 above)
Comment (nuevoalex) [reply packages/augur-core/source/contracts/reporting/ReputationToken.sol:40]: I'd prefer them being explicitly listed so they can be evaluated and we all see that.

**Pull request merged**